src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -636,6 +717,34 @@ public static void AppliedRequestFormLimits(this ILogger logger)
             }
         }
 
+        private static void ExecutingFilters(ILogger logger, string filterType, IEnumerable<IFilterMetadata> filters)
+        {
+            string[] filterList = null;
+            if (filters.Any())
+            {
+                filterList = GetFilterList(filters);
+            }
+
+            _executingFilters(logger, filterType, filterList, null);
+        }
+
+        private static string[] GetFilterList(IEnumerable<IFilterMetadata> filters)
+        {
+            var filterList = new List<string>();
```

Comment:
This code is now allocate-y. Does this data not already exist in some place where we don't need to do any additional computation in order to log it? In other words, can we get the list(s) of filters to show up in the log without doing any extra work?

Reply:
Discussed in-person with @kichalla :
1. This code doesn't get called unless this log level is enabled. And if this log level is enabled, perf ain't yo problem.
2. The whole idea of this part of the logging is to tell the developer in advance what the "filter execution plan" is. After the plan is logged, then the plan is executed, and you can see one-by-one as the filters run (or short-circuit, and thus *not* run).